torch/testing/_internal/opinfo/utils.py

```diff
@@ -258,3 +258,14 @@ def wrapper(x: np.ndarray, *args, **kwargs):
         return result
 
     return wrapper
+
+def numpy_prod_with_int64(): 
+    g = reference_reduction_numpy(np.prod)
+
+    @wraps(g)
+    def wrapper(x: np.ndarray, *args, **kwargs):
```

Comment:
Why wrap `g` here? Wouldn't `numpy_prod_with_int64` already be the wrapper?

Are there other operations with similar behavior? If so, I could see making this a factory, but that could also be left for the future